### PR TITLE
Adapt Nova usage of commitment to the generic trait

### PR DESCRIPTION
**Pull request by arnaucube**

### CODE REVIEW

File: src/folding/nova/decider.rs [hunk 1]
```diff
@@ -216,25 +220,33 @@ where
...
+impl<C1, GC1, C2, GC2, CP1, CP2> DeciderCircuit<C1, GC1, C2, GC2, CP1, CP2>
 where
     C1: CurveGroup,
     C2: CurveGroup,
     GC1: CurveVar<C1, CF2<C1>>,
     GC2: CurveVar<C2, CF2<C2>>,
+    CP1: CommitmentProver<C1>,
+    CP2: CommitmentProver<C2>,
+    // enforce that the CP2 is Pedersen commitment
+    PedersenParams<C2>: From<<CP2 as CommitmentProver<C2>>::Params>,
```
Comment (han0110) [inline src/folding/nova/decider.rs:232]: Perhaps just fix the `Params` directly? Or if there is a chance the param is in different struct then it's fine to keep as is.
```suggestion
    CP2: CommitmentProver<C2, Params = PedersenParams<C2>>,
    // enforce that the CP2 is Pedersen commitment
```

**Review by han0110 — APPROVED** [review]
LGTM! Just a tiny question.

File: src/folding/nova/decider.rs (hunk 1 above)
Comment (arnaucube) [reply src/folding/nova/decider.rs]: Oh right! [Updated](https://github.com/privacy-scaling-explorations/folding-schemes/compare/57180c01340c50ca73c96c3d95a1115f87443fb1..9590ad31bc05965959216a673ac139c986354387), thanks!

File: rust-toolchain [hunk 2]
```diff
@@ -1 +1 @@
-1.73.0
+1.75.0
```
[the remark this answers is not in the document]
Comment (arnaucube) [reply rust-toolchain:1]: I was keeping with the latest release, but if it makes sense to stick to a specific one I'm also happy.

**Pull request merged**